src/components/TextInput/BaseTextInput/isTextInputFocused.ts

```diff
@@ -0,0 +1,7 @@
+import type {AnimatedTextInputRef} from '@components/RNTextInput';
+import type {BaseTextInputRef} from './types';
+
+/** Checks that text input has the isFocused method and is focused. */
+export default function isTextInputFocused(textInput: React.MutableRefObject<BaseTextInputRef | null>): boolean | null {
+    return textInput.current && 'isFocused' in textInput.current && (textInput.current as AnimatedTextInputRef).isFocused();
+}
```

Comment:
This feels more like some utility function, is this indeed the best place to put such method based on our guidelines?

Reply:
This is the main/core text input so nab, moving this ahead